faunadb-scala/src/test/scala/faunadb/ClientSpec.scala

```diff
@@ -651,24 +651,54 @@ class ClientSpec
   }
 
   it should "test miscellaneous functions" in {
+    // NewId
     val newIdR = client.query(NewId()).futureValue
     newIdR.to[String].get should not be null
 
+    // Equals
     val equalsR = client.query(Equals("fire", "fire")).futureValue
     equalsR.to[Boolean].get shouldBe true
 
+    // Concat
     val concatR = client.query(Concat(Arr("Magic", "Missile"))).futureValue
     concatR.to[String].get shouldBe "MagicMissile"
 
     val concat2R = client.query(Concat(Arr("Magic", "Missile"), " ")).futureValue
     concat2R.to[String].get shouldBe "Magic Missile"
 
+    // Contains
     val containsR = client.query(Contains("favorites" / "foods", Obj("favorites" -> Obj("foods" -> Arr("crunchings", "munchings"))))).futureValue
     containsR.to[Boolean].get shouldBe true
 
     client.query(Contains("field", Obj("field" -> "value"))).futureValue shouldBe TrueV
     client.query(Contains(1, Arr("value0", "value1", "value2"))).futureValue shouldBe TrueV
 
+    client.query(Contains("a" / "nested" / 0 / "path", Obj("a" -> Obj("nested" -> Arr(Obj("path" -> "value")))))).futureValue shouldBe TrueV
+
+    // ContainsField
+    client.query(ContainsField("field", Obj("field" -> "value"))).futureValue shouldBe TrueV
+    client.query(ContainsField("field", Obj())).futureValue shouldBe FalseV
+
+    // ContainsPath
+    val containsPath = client.query(ContainsPath("favorites" / "foods", Obj("favorites" -> Obj("foods" -> Arr("crunchings", "munchings"))))).futureValue
+    containsPath.to[Boolean].get shouldBe true
+
+    client.query(ContainsPath("field", Obj("field" -> "value"))).futureValue shouldBe TrueV
+    client.query(ContainsPath(1, Arr("value0", "value1", "value2"))).futureValue shouldBe TrueV
+
+    client.query(ContainsPath("a" / "nested" / 0 / "path", Obj("a" -> Obj("nested" -> Arr(Obj("path" -> "value")))))).futureValue shouldBe TrueV
+
+    // ContainsValue
```

Comment:
According to [DRV-175](https://faunadb.atlassian.net/browse/DRV-175):

> `ContainsValue` can operate on refs, arrays, sets, and objects

I haven't created any test cases for refs and arrays. And, I think there's none for them in `core` either, right?

https://github.com/fauna/core/blob/479c9e1ee6d52d0d61f5899ae9da0ce1e71ae503/ext/api/src/test/scala/queries/BasicFunctionsSpec.scala#L1307

@nathaniel-may & @marrony, should we add them here? Is there any example of the expected behaviour for those?

Reply:
> I haven't created any test cases for refs and arrays. And, I think there's none for them in core either, right?
what do you mean? The very first 2 tests are for refs and arrays respectively

Reply:
Sorry Marrony, I meant _sets_ (not refs) and arrays...

So, from what I understood from the `core` specs, I've created two test cases for the Driver that should be analogous to the following:

```fql
// Contains Value on Ref
ContainsValue("269051745148076563", Ref(Collection("User"), "269051745148076563"))

// Contains Value on Object (or Document? Is it the same?)
ContainsValue(
  Ref(Collection("User"), "269051745148076563"),
  {
      "ref": Ref(Collection("User"), "269051745148076563"),
      "ts": 1592846589140000,
      "data": {}
    }
)
```

Could you confirm that those examples are covering the expected cases (`ContainsValue` on Ref and Object)? 

Also, could you maybe share examples (in plain FQL) for the Sets and Arrays cases (it should be easier to infer them from core! 🙂). Thank you!

Reply:
Never mind, Marrony! I just found the test cases for arrays, sets and objects in `core`, and I think I can pick up the necessary test cases from there. I guess that long spec in core tricked me at first!  Thank you!